Chore/dev override and docs (#10)
nearform/mira: docs/config/README.md

# Config File
This section describes how to configure the configuration file `./config/default.json`.

> Mechanism built-in into Mira is based on [node config](https://www.npmjs.com/package/config) library.

Reviewer: ```suggestion
Mira's configuration system is based on the [node config](https://www.npmjs.com/package/config) library.
```

## The Accounts Section

> The accounts section of the config file is an array of accounts that can be used for deploy apps and cicd. The accounts also represent environments.

Reviewer: ```suggestion
The accounts section of the config file is an array of account objects that can be used to deploy apps and run cicd. The accounts also represent environments.
```

        "region": "eu-west-1"
      },
      "profile": "mira-prod",
      "requireManualApproval": true
    }
  ]

}
```
## Accounts

> We define four accounts in the sample config file above: `Staging`, `Production`, `Default`, each with different account numbers.

Reviewer: Staging, Prod, and default would be 3 accounts unless I've missed one?

Author: I don't get it. Locally I have `We define three accounts`. Everything is pushed. Can you pull and check what will you see locally?

Domain management components (Certificate, Route53 and so on) are deployed on the `Domain` account. It uses `Z1234567890` as the Hosted Zone ID.

NB Example domain usage to follow in upcoming releases.

## Developer config

> To enable custom config modifications for developers working in the team, Mira expect a `config/dev.json` file to be created.

Reviewer: ```suggestion
To enable custom config modifications for developers working in the same team, Mira expects a `config/dev.json` file to be created.
```

__Note:__ This file should not by tracked in GIT.

> Whenever `config/dev.json` file is created and available in the runtime, contents of the `config/default.json` are going
> to be overridden in a shallow way. It means only top level properties are merged. The method used for this purpose 
> can be compared to [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for better understanding.

Reviewer: ```suggestion
Whenever a `config/dev.json` file is created and available at runtime, the contents of the `config/default.json` will be overridden in a shallow way. It means only the top-level properties are merged. The method used is comparable to [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).
```
